### PR TITLE
Mn 2017 07 export ideas

### PR DESCRIPTION
**Pull request by MagdaN**
First version for exporting ideas as xls, should be easy adaptable for documents. However if we want to make this really generic, we would probably need a custom url scheme for exports, where you pass in the module slug and then the view takes care of getting the right queryset. We can check with the other team - they loop through all phases and if within the phases view an explicit export view is set they use that one (I think)

Also, I am not sure about the permissions, who is allowed to pull that data?

### CODE REVIEW

**Comment by MagdaN** [comment]
I am using the code from the other team now - if we decide to move this to A4 we could also add the tests that they have written

File: euth/contrib/exports.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+        for rownum, row in enumerate(self.export_rows(), start=1):
+            for col, field in enumerate(row):
+                worksheet.write(rownum, col, field)
+
+        workbook.close()
+
+        return response
+
+
+class VirtualFieldMixin:
```
Comment (MagdaN) [inline euth/contrib/exports.py]: Yes, I copied it from the other team, I guess its about FK-Fields

File: euth/contrib/exports.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+    def get_queryset(self):
+        return super().get_queryset().filter(module=self.module)
+
+    def get_header(self):
+        return self._header
+
+    def export_rows(self):
+        for item in self.get_queryset().all():
+            yield [self.get_field_data(item, name) for name in self._names]
```
Comment (MagdaN) [inline euth/contrib/exports.py]: yes

File: euth/ideas/templatetags/idea_tags.py [hunk 3]
```diff
@@ -11,3 +13,8 @@ def get_range(number, listcount):
         return range(listcount-4, listcount+1)
     else:
         return range(number-2, number+3)
+
+
+@register.assignment_tag
+def is_idea_list(module):
+    return Idea.objects.filter(module=module).count() > 0
```
Comment (MagdaN) [inline euth/ideas/templatetags/idea_tags.py]: but I would also only add an download button if there are any ideas to download

**Comment by slomo** [comment]
Okay, so I shall merge this branch now?

**Comment by MagdaN** [comment]
yes if you like

**Pull request merged**